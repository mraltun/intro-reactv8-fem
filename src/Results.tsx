// We could have named the Pet component and the Pet interface differently (and this is where calling it IPet could have been useful) but it's good to see how to handle a collision like this. Just use as to import it as a different name.
import { Pet as PetType } from "./APIResponsesTypes";
import Pet from "./Pet";

const Results = ({ pets }: { pets: PetType[] }) => {
  return (
    <div className="grid gap-4 grid-cols-1 sm:grid-cols-2 lg:grid-cols-3">
      {!pets.length ? (
        <h1>No Pets Found</h1>
      ) : (
        pets.map((pet) => (
          <Pet
            // {...pet}
            animal={pet.animal}
            id={pet.id}
            name={pet.name}
            breed={pet.breed}
            images={pet.images}
            location={`${pet.city}, ${pet.state}`}
            key={pet.id}
          />
        ))
      )}
    </div>
  );
};

export default Results;
