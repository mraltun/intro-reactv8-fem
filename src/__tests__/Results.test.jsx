// Snapshot tests are low confidence, low cost ways of writing tests. With more-or-less a single line of code you can assert: this code doesn't break, and it isn't changing over time.
import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import Results from "../Results";

test("renders correctly with no pets", () => {
  const { asFragment } = render(<Results pets={[]} />);
  expect(asFragment()).toMatchSnapshot();
});
