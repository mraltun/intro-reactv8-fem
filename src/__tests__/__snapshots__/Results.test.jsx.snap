// Vitest Snapshot v1

exports[`renders correctly with no pets 1`] = `
<DocumentFragment>
  <div
    class="grid gap-4 grid-cols-1 sm:grid-cols-2 lg:grid-cols-3"
  >
    <h1>
      No Pets Found
    </h1>
  </div>
</DocumentFragment>
`;
